lib/index.js

```diff
@@ -203,12 +234,24 @@ api.set = async ({
   } else {
     query['account.email'] = email;
   }
-  const result = await database.collections.account.updateMany(
-    query, {
-      $set: {
-        [`meta.bedrock-authn-token.tokens.${type}`]: token
-      }
-    }, database.writeOptions);
+
+  let result;
+  if(type === 'nonce') {
+    result = await database.collections.account.updateMany(
+      query, {
+        $push: {
+          [`meta.bedrock-authn-token.tokens.${type}`]: token
+        }
+      }, database.writeOptions);
+  } else {
+    result = await database.collections.account.updateMany(
+      query, {
+        $set: {
+          [`meta.bedrock-authn-token.tokens.${type}`]: token
+        }
+      }, database.writeOptions);
+  }
+
   if(result.result.n === 0) {
```

Comment:
This could simplify to:

```suggestion
  const updateType = type === 'nonce' ? '$push': '$set';
  const result = await database.collections.account.updateMany(
    query, {
      [updateType]: {[`meta.bedrock-authn-token.tokens.${type}`]: token}
    }, database.writeOptions);
  if(result.result.n === 0) {
```